### DISCUSSION TITLE
How to know if the audio playing has reached its end or stopped?

### DISCUSSION THREAD

**Discussion by readonlymem0ry** [pschatzmann/arduino-audio-tools#1545] [Q&A] [answered]
How to know and how to stop the copier or copy() when the audio playing has reached its end or stopped?

**Comment by pschatzmann** [comment]
Just read the Wiki: [The most important basics!](https://github.com/pschatzmann/arduino-audio-tools/wiki/Introduction)
or the class documentation

**Reply by readonlymem0ry** [reply]
Can you give me a shortcut to more detailed documentation of what I mean?

here is my code:

while (Serial.available() > 0) {
    String inputString = Serial.readStringUntil('\n');
      url.begin(inputString.c_str());
      for(int x; x > 0; x++){
        copier.copy();
    }
  }

i use "for" looping to run the copier, but I don't know how many iterations it takes, is there a solution to my problem?

**Reply by pschatzmann** [reply, marked as answer]
You still did not read the Wiki! You can find the answer there in the last sentence of the Copy Section!
I spent quite some time to put together the documentation to avoid these kind of unnecessary questions
